diff --git a/src/com/android/imsserviceentitlement/ImsEntitlementApi.java b/src/com/android/imsserviceentitlement/ImsEntitlementApi.java
index 74943ac..926497d 100644
--- a/src/com/android/imsserviceentitlement/ImsEntitlementApi.java
+++ b/src/com/android/imsserviceentitlement/ImsEntitlementApi.java
@@ -29,8 +29,6 @@ import androidx.annotation.VisibleForTesting;
 import com.android.imsserviceentitlement.entitlement.EntitlementConfiguration;
 import com.android.imsserviceentitlement.entitlement.EntitlementConfiguration.ClientBehavior;
 import com.android.imsserviceentitlement.entitlement.EntitlementResult;
-import com.android.imsserviceentitlement.fcm.FcmTokenStore;
-import com.android.imsserviceentitlement.fcm.FcmUtils;
 import com.android.imsserviceentitlement.ts43.Ts43Constants.ResponseXmlAttributes;
 import com.android.imsserviceentitlement.ts43.Ts43Constants.ResponseXmlNode;
 import com.android.imsserviceentitlement.ts43.Ts43SmsOverIpStatus;
@@ -104,8 +102,8 @@ public class ImsEntitlementApi {
         ServiceEntitlementRequest.Builder requestBuilder = ServiceEntitlementRequest.builder();
         mLastEntitlementConfiguration.getToken().ifPresent(
                 token -> requestBuilder.setAuthenticationToken(token));
-        FcmUtils.fetchFcmToken(mContext, mSubId);
-        requestBuilder.setNotificationToken(FcmTokenStore.getToken(mContext, mSubId));
+        //FcmUtils.fetchFcmToken(mContext, mSubId);
+        //requestBuilder.setNotificationToken(FcmTokenStore.getToken(mContext, mSubId));
         // Set fake device info to avoid leaking
         requestBuilder.setTerminalVendor("vendorX");
         requestBuilder.setTerminalModel("modelY");
